src/main/java/sirius/kernel/nls/NLS.java

```diff
@@ -1025,14 +1027,14 @@ public static <V> V parseMachineString(Class<V> clazz, String value) {
     private static <V> V parseBasicTypesFromMachineString(Class<V> clazz, String value) {
         if (Integer.class.equals(clazz) || int.class.equals(clazz)) {
             try {
-                return (V) Integer.valueOf(value);
+                return (V) extractIntegerFromDouble(() -> parseMachineString(Double.class, value));
             } catch (NumberFormatException e) {
                 throw new IllegalArgumentException(fmtr("NLS.errInvalidNumber").set("value", value).format(), e);
             }
         }
         if (Long.class.equals(clazz) || long.class.equals(clazz)) {
             try {
-                return (V) Long.valueOf(value);
+                return (V) extractLongFromDouble(() -> parseMachineString(Double.class, value));
```

Comment:
can't we use new BigDecimal(value).longValueExact() and catch the ArtihmeticException below?